.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,62 @@
+name: E2E test on DevNet
+
+# Only trigger, when the build workflow succeeded, and allow manual triggering.
+on:
+  workflow_dispatch:
+  workflow_run:
+    workflows: ["Test, build and push artifacts"]
+    types:
+      - completed
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'e2e-devnet-test')
+    env:
+      ARGO_SERVER: "workflows.dev-us-east4-1.poktnodes.network:8443"
+      ARGO_HTTP1: true
+      ARGO_SECURE: true
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    steps:
+      - id: "auth"
+        uses: "google-github-actions/auth@v1"
+        with:
+          credentials_json: "${{ secrets.ARGO_WORKFLOW_EXTERNAL }}"
+
+      - id: "get-credentials"
+        uses: "google-github-actions/get-gke-credentials@v1"
+        with:
+          cluster_name: "nodes-gcp-dev-us-east4-1"
+          location: "us-east4"
+
+      - id: "install-argo"
```

Comment:
Do github actions always require a clean slate?

Reply:
@Olshansk GitHub actions always start with a clean state, it is up to a user to checkout the repo, install dependencies, etc. However, there is a way to cache the binaries. I don't think it makes a difference to use cache or download a single binary from GitHub, so we do that here every time.

Reply:
Understood, thanks for the explanation.